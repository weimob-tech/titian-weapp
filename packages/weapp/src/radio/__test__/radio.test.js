const simulate = require('miniprogram-simulate');
const { getComponentId } = require('../../common/test/getComponent');

describe('test component', () => {
  const radio = getComponentId('radio');
  it('should defaultValue setting success', async () => {
    const onChange = jest.fn();
    const container = simulate.render(
      simulate.load({
        usingComponents: {
          'ti-radio': radio
        },
        template: `
          <ti-radio id='radio' bind:change='onChange' value='1' checked='{{checked}}' defaultChecked='{{defaultChecked}}'>a</ti-radio>
        `,
        data: {
          defaultChecked: false,
          checked: null
        },
        methods: {
          onChange
        }
      })
    );

    container.attach(document.createElement('parent-wrapper'));
    await simulate.sleep(40);
    expect(container.toJSON()).toMatchSnapshot();
    const radioIconWrap = container.querySelector('#radio').querySelector('.titian-radio-icon-wrap');
    const radioLabelWrap = container.querySelector('#radio').querySelector('.titian-radio-label');

    {
      radioIconWrap.dispatchEvent('tap');
      await simulate.sleep(40);
      expect(container.toJSON()).toMatchSnapshot();
      expect(onChange).toBeCalledTimes(1);

      radioLabelWrap.dispatchEvent('tap');
      await simulate.sleep(40);
      expect(onChange).toBeCalledTimes(2);
    }
  });

  it('should checked control component success', async () => {
    const onChange = jest.fn();
    const container = simulate.render(
      simulate.load({
        usingComponents: {
          'ti-radio': radio
        },
        template: `
          <ti-radio id='radio' value='1' checked='{{checked}}'>a</ti-radio>
        `,
        data: {
          checked: false
        },
        methods: {
          onChange
        }
      })
    );

    container.attach(document.createElement('parent-wrapper'));
    await simulate.sleep(40);
    expect(container.toJSON()).toMatchSnapshot();

    const radioIconWrap = container.querySelector('#radio').querySelector('.titian-radio-icon-wrap');

    {
      radioIconWrap.dispatchEvent('tap');
      await simulate.sleep(40);
      expect(onChange).not.toBeCalled();

      expect(
        container.querySelector('#radio').querySelector('.titian-radio').dom.className.includes('titian-radio-checked')
      ).toBe(false);
    }

    {
      container.setData({
        checked: true
      });
      radioIconWrap.dispatchEvent('tap');
      await simulate.sleep(40);
      expect(onChange).not.toBeCalled();

      expect(
        container.querySelector('#radio').querySelector('.titian-radio').dom.className.includes('titian-radio-checked')
      ).toBe(true);
    }
  });
});
