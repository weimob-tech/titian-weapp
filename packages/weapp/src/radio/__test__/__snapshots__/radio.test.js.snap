// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test component should checked control component success 1`] = `
<main>
  <main
    id="radio"
  >
    <wx-view
      class="ext-class titian-radio-horizontal titian-radio"
    >
      <wx-view
        class="titian-radio-icon-wrap"
        style="--base-radius-size:999px; height: 32rpx"
        bind:tap="onChange"
      >
        <ti-icon
          extClass="icon-class titian-radio-icon-animation titian-radio-icon"
        >
          <wx-view
            class="titian-icon-base titian-icon titian-icon-checkbox-hollow ext-class"
            style="line-height:32rpx;height:32rpx;width:32rpx;font-size:32rpx"
            bind:tap="onClick"
          />
        </ti-icon>
      </wx-view>
      <wx-view
        class="titian-radio-label"
        bind:tap="onClick"
      >
        a
      </wx-view>
    </wx-view>
  </main>
</main>
`;

exports[`test component should defaultValue setting success 1`] = `
<main>
  <main
    id="radio"
    bind:change="onChange"
  >
    <wx-view
      class="ext-class titian-radio-horizontal titian-radio"
    >
      <wx-view
        class="titian-radio-icon-wrap"
        style="--base-radius-size:999px; height: 32rpx"
        bind:tap="onChange"
      >
        <ti-icon
          extClass="icon-class titian-radio-icon-animation titian-radio-icon"
        >
          <wx-view
            class="titian-icon-base titian-icon titian-icon-checkbox-hollow ext-class"
            style="line-height:32rpx;height:32rpx;width:32rpx;font-size:32rpx"
            bind:tap="onClick"
          />
        </ti-icon>
      </wx-view>
      <wx-view
        class="titian-radio-label"
        bind:tap="onClick"
      >
        a
      </wx-view>
    </wx-view>
  </main>
</main>
`;

exports[`test component should defaultValue setting success 2`] = `
<main>
  <main
    id="radio"
    bind:change="onChange"
  >
    <wx-view
      class="ext-class titian-radio-checked titian-radio-horizontal titian-radio"
    >
      <wx-view
        class="titian-radio-icon-wrap-checked titian-radio-icon-wrap"
        style="--base-radius-size:999px; height: 32rpx"
        bind:tap="onChange"
      >
        <ti-icon
          extClass="icon-class titian-radio-icon-zoom-in titian-radio-icon-checked titian-radio-icon-animation titian-radio-icon"
        >
          <wx-view
            class="titian-icon-base titian-icon titian-icon-checkbox-hollow ext-class"
            style="line-height:32rpx;height:32rpx;width:32rpx;font-size:32rpx"
            bind:tap="onClick"
          />
        </ti-icon>
      </wx-view>
      <wx-view
        class="titian-radio-label"
        bind:tap="onClick"
      >
        a
      </wx-view>
    </wx-view>
  </main>
</main>
`;
